src/main/java/de/htwberlin/GradesOverview.java

```diff
@@ -1,27 +1,115 @@
 package de.htwberlin;
 
-public class GradesOverview {
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+import java.util.Arrays;
 
+public class GradesOverview {
   private int[] grades;
+  private int grade ;
+
+
+
+
 
+
+  /**
+   * method to initialize an array with size of 10
+   */
   public GradesOverview() {
-    grades = new int[5];
+    grades = new int[10];
+    // for (int x=0; x < grades.length; x++){
+    // if (x == grades.length - 1 && grades[grades.length-1] != 0) {}
+
   }
 
-  public int[] getGrades() {
-    return grades;
+  /**
+   * gitter for the grades
+   *
+   * @return the grades in an array
+   */
+  public int[] getGrades() { return grades; }
+  public int getGrade() {
+    return grade;
   }
 
   public void add(int grade) {
     // TODO
+    if (grade > 5 || grade <=0) {
+      throw new IllegalArgumentException("nop");
+    }
+    for (int i = 0; i < grades.length; i++) {
+      if (grades[i] == 0) {
+        grades[i] = grade;
+        break;
+      }
+
+    }
+    if ( grades[grades.length -1] != 0){
+      grades = Arrays.copyOf(grades, grades.length + 1);
+      if (grades.length >25){
+        grades = Arrays.copyOf(grades, grades.length  -1);
+      }
+
+      for (int x: grades){
+        System.out.println(x);}};
+    System.out.println(grade);
+    this.grade = grade;
+    calcAverage();
   }
 
+
   double calcAverage() {
-    return 0.0; // TODO return the calculated average
+    DecimalFormat df2 = new DecimalFormat("#.##");
+    double n = 0d;
+    double r = 0d;
+    int f =0;
+    for (int i = 0; i < grades.length; i++) {
+      if(getGrades()[i] !=0){
+        n += getGrades()[i];
+        f++;}
+    }
+    r = n / f;
+    if (n == 0){
+      r= 0.00;
+    }
+    BigDecimal bd = new BigDecimal(r).setScale(2, RoundingMode.HALF_UP);
+    double newInput = bd.doubleValue();
+    return newInput; // TODO return the calculated average*/
```

Comment:
Versuche immer mit sprechenden Variablennamen zu arbeiten, damit der Code leichter verständlich wird. Die Rundungsfunktion könnte bspw. auch eine eigene Methode sind.

Reply:
Danke für die Hinweis ich werde in den kommenden Übungen  drauf achten